src/core/WaveContract.sol

```diff
@@ -92,126 +103,129 @@ contract WaveContract is ERC2771Context, Ownable, ERC721, SignatureVerifier, IWa
         startTimestamp = _startTimestamp;
         endTimestamp = _endTimestamp;
         isSoulbound = _isSoulbound;
+        tokenRewards = _tokenRewards;
 
-        claimRewardsLength = uint8(_claimRewards.length);
-        for (uint8 i = 0; i < claimRewardsLength; ++i) {
-            claimRewards.push(_claimRewards[i]);
-        }
-
-        raffleRewardsLength = uint8(_raffleRewards.length);
-        for (uint8 i = 0; i < raffleRewardsLength; ++i) {
-            raffleRewards.push(_raffleRewards[i]);
-        }
+        if (_tokenRewards.token != address(0)) isERC20Campaign = true;
+        mintsPerClaim = 1;
+        shouldVerifySignature = true;
     }
 
-    /// @notice Allows the governance and the owner of the contract to set metadata base URI for all tokens
-    /// @param _uri The base URI to set
-    /// @param _customMetadata Whether the metadata is encoded with rewardId or tokenId
-    function changeBaseURI(string memory _uri, bool _customMetadata) public onlyAuthorized {
+    /// @inheritdoc IWaveContract
+    function changeBaseURI(string calldata _uri, bool _customMetadata) public onlyAuthorized {
         _metadataBaseURI = _uri;
         customMetadata = _customMetadata;
     }
 
-    /// @notice Allows the owner to set the campaign start timestamp
-    function setStartTimestamp(uint256 _startTimestamp) public onlyOwner {
+    /// @inheritdoc IWaveContract
+    function setStartTimestamp(uint256 _startTimestamp) public onlyAuthorized {
         require(block.timestamp < _startTimestamp && _startTimestamp < endTimestamp, "Invalid start timestamp");
         startTimestamp = _startTimestamp;
     }
 
-    /// @notice Allows the governance to set the campaign end timestamp
-    function setEndTimestamp(uint256 _endTimestamp) public onlyOwner {
-        require(raffleRewardsLength == 0, "Can't change end timestamp for raffles");
+    /// @inheritdoc IWaveContract
+    function setEndTimestamp(uint256 _endTimestamp) public onlyAuthorized {
         require(_endTimestamp > block.timestamp && _endTimestamp > startTimestamp, "Invalid end timestamp");
         endTimestamp = _endTimestamp;
     }
 
-    /// @notice Allows the owner to end the campaign early
-    function endCampaign() public onlyActive onlyOwner {
+    /// @dev toggle the `shouldVerifySignature` boolean
+    function toggleVerifySignature() public onlyGovernance {
+        shouldVerifySignature = !shouldVerifySignature;
+    }
+
+    /// @dev toggle the `isERC20Campaign` boolean
+    function toggleIsERC20Campaign() public onlyGovernance {
+        isERC20Campaign = !isERC20Campaign;
+    }
```

Comment:
As a personal preference, I like more having a setter method instead of a toggle. This assures that we always set the value intended to avoid manual mistakes.